File: 2015-2016/Projetos/project_java/src/project_java/tp1/comport/Reaccao.java
package project_java.tp1.comport;

public class Reaccao implements Comportamento{
	
	private Estimulo estimulo;
	private Accao accao;
	
	public Reaccao(Estimulo estimulo, Accao accao) {
		this.estimulo = estimulo;
		this.accao = accao;
	}
	
	public Accao activar(Estimulo estimulo){
		if( estimulo == this.estimulo){
			return this.accao;
		}
		return null;
	}

}

File: 2018-2019/Trabalhos Práticos/iasa_cp3/iasa_cp3/src/lib/ecr/reaccao.py
'''
Created on 12/04/2019

@author: Ana Oliveira
'''
from ecr.comportamento import Comportamento


class Reaccao(Comportamento):


    def _detectar_estimulo(self, percepcao):
        raise NotImplementedError("Abstract")
        
        
    def _gerar_resposta(self, estimulo):
        raise NotImplementedError("Abstract")
        
        
    def activar(self, percepcao):
        estimulo = self._detectar_estimulo(percepcao)
        
        if ((estimulo is not None) and (estimulo is not False)):
            return self._gerar_resposta(estimulo)
        return None